### DISCUSSION TITLE
Are there any metric to determine a players overall strength?

### DISCUSSION THREAD

**Discussion by Mr-Bajs** [ihhub/fheroes2#8186] [General]
I'm just wondering if there is any metric to determine a players strength? If not would it be possible to make one? I guess it would be hard asses what each data point is worth weighing that to another data point. But would it be reasonable to convert all of a player assets resources, towns, castle, building, creatures and heroes into gold, hence gold is the most important resource?

I'm mainly asking is beacuse it would be useful if you could track players overall strength when playtesting maps.

And how is the metric Total Army Strength in the thieves guild measured?

**Comment by zenseii** [comment]
Hi, @Mr-Bajs.

I'm not sure I understand what you mean. Are you requesting a way to compare all the statistics for the kingdoms in one unified metric?

I don't see how that could be useful when you can easily place an Oracle building and view all the various metrics that would compose such an "overall strength".

Ex. In some maps you need to accumulate most gold, so the most important metric is total gold in the treasury in the Oracle, and not the Total Army Strength.

Therefore, you can judge a player's progress much better this way.

I don't remember right now how Total Army Strength is calculated.

**Comment by Mr-Bajs** [comment]
Yes one unified metric i guess what is was going for, but then thinking about it...

I mostly think of a usecase for this as a logfile that maybe logs all kingdoms data for each day, then you can track each kingdoms progress. Then of course a unified metric isn't needed, because if the data is accessible any mapmaker choose what metrics are useful for whatever map he or she creates. It if where a simple logfile then anyone could use that in Excel or LibreCalc for statistical analasis. 

Not requesting anything, just thinking broadly. When creating a map it would be pretty useful if AI vs AI only playtesting would output some datapoints in a logfile or so. Doing 20 AI vs AI playtesting, that could give an logfile output that player Purple wins 70% of the time, might give an indication that the Purple player is overpowerd (unless it's intended). Or that player Blue is first eradicated player 50% of the playtests, indicating it's have a weak starting postion etc. Leading to less guesswork for the mapmaker.
